libraries/plugins/chain/chain_plugin.cpp

```diff
@@ -476,9 +476,9 @@ void chain_plugin::plugin_startup()
    {
       database_config = fc::json::from_file( my->database_cfg, fc::json::strict_parser );
    }
-   catch ( const std::exception& e )
+   catch ( ... )
```

Comment:
Let's add catching an fc::assert_exception so that

1) We can log it
2) We don't suppress unexpected exception types (i.e. boost exceptions) as a database configuration error.